### ISSUE TITLE
Support registering to globalThis

### ISSUE THREAD

**Issue by anuraaga** [gopherjs/gopherjs#1316]
I am using a gopherjs binary with [QuickJS](https://github.com/bellard/quickjs), which uses `globalThis` for its global. I wonder if it's possible to support `globalThis` in the prelude to work out of the box with QuickJS.

https://github.com/gopherjs/gopherjs/blob/cfa0783a7b031caf2220882339fa14d879ece43e/compiler/prelude/prelude.js#L17

```js
if (typeof window !== "undefined") { /* web page */
    $global = window;
} else if (typeof self !== "undefined") { /* web worker */
    $global = self;
} else if (typeof global !== "undefined") { /* Node.js */
    $global = global;
    $global.require = require;
} else if (typeof globalThis !== "undefined") { /* QuickJS */
  $global = globalThis;
} else { /* others (e.g. Nashorn) */
    $global = this;
}
```

It's not difficult to shim though so not a big deal either way.

https://github.com/wasilibs/go-prettier/blob/main/buildtools/wasm/global.ts#L1
`(globalThis as any).self = globalThis;`

**Comment by nevkontakte** [comment]
Yeah, that seems reasonable. If you don't mind sending a PR, I'll be happy to merge it :)

**Comment by BrownNPC** [comment]
Any updates @anuraaga
